spec/abstractops.html

```diff
@@ -4,6 +4,29 @@
 <emu-clause id="sec-temporal-abstract-ops">
   <h1>Abstract operations</h1>
 
+  <!-- Based on ECMA-262 IterableToList -->
+  <emu-clause id="sec-iterabletolistoftype" aoid="IterableToListOfType">
+    <h1>IterableToList ( _items_, _elementTypes_ )</h1>
+    <p>
+      The abstract operation IterableToListOfType takes arguments _items_ and _elementTypes_ (a List of names of ECMAScript Language Types).
+      It is used to create a List value whose elements are provided by the values of the iterable _items_.
+      _elementTypes_ contains the names of ECMAScript Language Types that are allowed for element values of the List that is created.
+      It performs the following steps when called:
+    </p>
+    <emu-alg>
+      1. Let _iteratorRecord_ be ? GetIterator(_items_, ~sync~).
+      1. Let _values_ be a new empty List.
+      1. Let _next_ be *true*.
+      1. Repeat, while _next_ is not *false*,
+        1. Set _next_ to ? IteratorStep(_iteratorRecord_).
+        1. If _next_ is not *false*, then
+          1. Let _nextValue_ be ? IteratorValue(_next_).
+          1. If Type(_nextValue_) is not an element of _elementTypes_, throw a *TypeError* exception.
```

Comment:
I believe you should close the iterator before throwing, as in

```suggestion
          1. If Type(_nextValue_) is not an element of _elementTypes_, then
            1. Let _completion_ be ThrowCompletion(a newly created TypeError object).
            1. Return ? IteratorClose(_iteratorRecord_, _completion_).
```